modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/util/CTCollectionTestUtil.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.change.tracking.internal.util;
+
+import com.liferay.change.tracking.model.CTCollection;
+import com.liferay.change.tracking.service.CTCollectionLocalService;
+import com.liferay.change.tracking.service.CTCollectionService;
+import com.liferay.journal.model.JournalArticle;
+import com.liferay.journal.service.JournalArticleLocalService;
+import com.liferay.petra.lang.SafeCloseable;
+import com.liferay.portal.kernel.change.tracking.CTCollectionThreadLocal;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LogEntry;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import org.junit.Assert;
+
+import java.util.List;
+
+public class CTCollectionTestUtil {
+
+	public static CTCollection addCTCollection(User user)
+		throws Exception {
+
+		CTCollection ctCollection = _ctCollectionLocalService.addCTCollection(
+			null, TestPropsValues.getCompanyId(), user.getUserId(), 0,
+			RandomTestUtil.randomString(), null);
+
+		return ctCollection;
+	}
+
+	public static void publishCTCollectionWithConflicts(
+		CTCollection ctCollection, Group group, JournalArticle journalArticle)
+	throws Exception {
+		try (SafeCloseable safeCloseable =
+				 CTCollectionThreadLocal.setCTCollectionIdWithSafeCloseable(
+					 ctCollection.getCtCollectionId())) {
+
+			journalArticle = _journalArticleLocalService.updateArticle(
+				journalArticle.getId(), RandomTestUtil.randomString());
+		}
+
+		_journalArticleLocalService.deleteArticle(
+			group.getGroupId(), journalArticle.getArticleId(),
+			ServiceContextTestUtil.getServiceContext());
+
+		publishCTCollection(ctCollection.getCtCollectionId());
+	}
+
+	public static void publishCTCollection(long ctCollectionId) throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+			"com.liferay.portal.background.task.internal.messaging." +
+			"BackgroundTaskMessageListener",
+			LoggerTestUtil.ERROR)) {
+
+			_ctCollectionService.publishCTCollection(
+				TestPropsValues.getUserId(), ctCollectionId);
+
+			List<LogEntry> logEntries = logCapture.getLogEntries();
+
+			Assert.assertEquals(logEntries.toString(), 1, logEntries.size());
+
+			LogEntry logEntry = logEntries.get(0);
+
+			Assert.assertEquals(
+				"Unable to execute background task", logEntry.getMessage());
+		}
+	}
```

Comment:
This should be defined inside `publishCTCollectionWithConflicts` to make it clear that this publish method should only be used when conflicts are expected. This will lead to assertion errors if it's used to publish publications without conflicts.